n3fit/src/n3fit/performfit.py

```diff
@@ -9,11 +9,74 @@
 import numpy as np
 
 import n3fit.checks
+
+from collections.abc import Callable
 from n3fit.vpinterface import N3PDF
+from validphys.loader import Loader
+from validphys.pdfbases import evolution
 
 log = logging.getLogger(__name__)
 
 
+def wrap_lhapdf(pdfset_name: str, q0value: float) -> Callable:
+    """
+    Wrapper around LHAPDF to compute the PDF predictions in the
+    EVOLUTION basis given an array of x values. The Q0 value is
+    inferred from the fitting scale input from the run card.
+
+    Parameters
+    ----------
+    pdfset_name: str
+        name of the PDF set, input as `unpolpdf` from the runcard
+    q0value: float
+        value of Q2 at the fitting scale
+
+    Returns
+    -------
+    callable:
+        a callable that evaluate the PDF for a given x, when
+        called it returns a array of shape (n_xgrid, npids)
+
+    """
+    # TODO: Move this to the Provider and Add checks
+    pdfset = Loader().check_pdf(pdfset_name)
+    # Include all the PDF flavour in the order of FKs
+    pids = [
+        'photon',
+        'singlet',
+        'gluon',
+        'v',
+        'v3',
+        'v8',
+        'v15',
+        'v24',
+        'v35',
+        't3',
+        't8',
+        't15',
+        't24',
+        't35',
+    ]
+
+    def compute_asx(xgrid: list) -> np.ndarray:
+        """
+        Compute the PDF predictions for a given x and PID values
+
+        Parameters
+        ----------
+        xgrid: list
+            list of xgrid values to compute predictions from
+
+        """
+        # `res` is of shape (n_replicas, n_fl=14, n_x, n_q2=1)
+        res = evolution.grid_values(pdfset, pids, xgrid, [q0value])
+        # TODO: for the time being select the Central replicas,
+        # but modify the following to also returns the STD
+        return np.squeeze(res[0].swapaxes(0, -1))  # Shape=(nx, n_fl=14)
+
+    return compute_asx
```

Comment:
Please, use directly validphys for this. You can get results from validphys in whatever basis you want.

Reply:
Yes, that'd be indeed the plan as per the todo. Once we are confident that this is the conceptual approach to enforce positivity I'll do so.